### PR TITLE
refactor/nfsv4 acl compat

### PR DESCRIPTION
**Pull request by fightingtexasaggie**
To test:

The NetApp is already connected to griidc-dev, so this branch needs to be merged into anything else you're working on if you need files to work.

1. create/approve new dif, then submit a dataset with a small file to test.
2. Approve the submission, then download it from dataland.
3. Repeat for http pull and sftp uploaded file.

Check permissions on:
/san/data/store/FOLDER
/san/data/store/FOLDER/DATAFILE
/san/data/download/FOLDER
/san/data/download/FOLDER/DATAFILE

In order to see the NFS ACL permissions, use the nfs4_getfacl command.
pertinent info: 48 = UID of 'apache' user, 1008 = GID of 'griidc' ldap group.
r=read, x=execute (not traversal as with UNIX), t = traversal

### CODE REVIEW

**Comment by fightingtexasaggie** [comment]
Let's keep this out of the Develop branch until we have a deployment strategy figured out.

File: src/Pelagos/Util/DataStore.php [hunk 1]
```diff
@@ -424,4 +410,26 @@ protected function getStoreFileName($datasetId, $type)
...
+     * @param boolean $isGroup Flag set to true if a group, defaults to user.
+     *
+     * @return string|boolean
+     */
+    protected function getIdFromName(string $name, bool $isGroup = false)
+    {
+        $id = false;
+        if ($isGroup) {
+            // This call returns false on failure to resolve.
+            $obj = posix_getgrnam($name);
```
Comment (praneethpr) [inline src/Pelagos/Util/DataStore.php]: Is ext-posix required to be added to composer.json?

File: src/Pelagos/Util/DataStore.php [hunk 2]
```diff
@@ -424,4 +410,26 @@ protected function getStoreFileName($datasetId, $type)
...
+            $id = $obj['gid'];
+        } else {
+            $obj = posix_getpwnam($name);
+            $id = $obj['uid'];
+        }
+        return $id;
```
Comment (praneethpr) [inline src/Pelagos/Util/DataStore.php]: Can id also have a string value?

File: src/Pelagos/Util/DataStore.php (hunk 1 above)
Comment (fightingtexasaggie) [reply src/Pelagos/Util/DataStore.php]: No, this is the built-in PHP function.

File: src/Pelagos/Util/DataStore.php (hunk 2 above)
Comment (fightingtexasaggie) [reply src/Pelagos/Util/DataStore.php]: no, just either a numeric value or a boolean false as posix_getpwnam and the other return false on failure. Come to think of it, it's probably better I throw an exception if that were to ever happen.

Comment (praneethpr) [reply src/Pelagos/Util/DataStore.php]: Oh then, we should probably check if array exists before trying to access the array $obj

File: app/config/config_drupal_dev.yml [hunk 3]
```diff
@@ -10,6 +10,9 @@ framework:
         strict_requirements: true
     profiler: { only_exceptions: false }
 
+fos_js_routing:
+    request_context_base_url: /pelagos-symfony
```
Comment (mickel1138) [inline app/config/config_drupal_dev.yml]: I thought this was already in develop?

Comment (mickel1138) [reply app/config/config_drupal_dev.yml]: If you need it for _your_ development enviroment, add .local to that yaml file.

**Pull request merged**